tests/nucleo-g474.rs

```diff
@@ -91,27 +86,28 @@ mod tests {
         let gpioa = dp.GPIOA.split(&mut rcc);
         let _pa1_important_dont_use_as_output = gpioa.pa1.into_floating_input();
         let mut pin = gpioa.pa8.into_open_drain_output();
+        let pin_num = 8; // PA8
 
         // Enable pull-up resistor
         {
             let gpioa = unsafe { &*GPIOA::PTR };
-            gpioa.pupdr().modify(|_, w| w.pupdr8().pull_up());
+            gpioa.pupdr().modify(|_, w| w.pupdr(pin_num).pull_up());
         }
 
         pin.set_high();
         delay.delay(1.millis()); // Give the pin plenty of time to go high
         assert!(pin.is_high());
         {
             let gpioa = unsafe { &*GPIOA::PTR };
```

Comment:
@burrbull Are we allowed to keep around this reference so its lifetime overlaps with the usage of `pin`? I added the scope here to avoid that but if allowed then we could create the reference once and keep that instead of all the blocks